src/Chapter4.hs

```diff
@@ -673,7 +724,62 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a = None | Node a (Tree a) (Tree a) deriving (Eq, Show)
+
+instance Functor Tree where
+    fmap _ None = None
+    fmap f (Node v l r) = Node (f v) (fmap f l) (fmap f r)
+
+instance Semigroup (Tree a) where
+    (<>) a b = fromList (toList a ++ toList b)
+    -- since we aren't actually defining any real structure for this tree
+
+instance Foldable Tree where
+    foldMap _ None = mempty
+    foldMap f (Node v l r) = f v `mappend` foldMap f l `mappend` foldMap f r
+
+instance Monoid (Tree a) where
+    mempty = None
+    mappend = (<>)
+
+toList :: Tree a -> [a]
+toList None = []
+toList (Node v l r) = toList l ++ (v : toList r)
+
+
+fromList :: [a] -> Tree a
+fromList [] = None
+fromList [x] = Node x None None
+fromList (x:xs) = let (l, r) = scramble xs
+                   in Node x (fromList l) (fromList r)
```

Comment:
If you want to make the task a bit more interesting, you can try to implement Binary Search Tree 🙂 
Otherwise, great work! 👍🏼

Reply:
I did! The shuffle function is to try and mix up the input list a bit in case it's already sorted, in which case a BST would just degenerate to a linked list. Implemented BST in 37a3392.